src/components/lib/posix/posix.c

```diff
@@ -134,9 +126,14 @@ cos_mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset)
 		assert(0);
 	}
 
-	void *ret = do_mmap(length);
+	addr = (void *)cos_page_bump_allocn(&parent_cinfo, length);
+	if (!addr){
+		ret = (void *) -1;
+	} else {
+		ret = addr;
+	}
 
-	if (ret == (void *)-1) { /* return value comes from man page */
+	if (ret == (void *)-1) {  //return value comes from man page
```

Comment:
It seems odd to have a condition previously that sets the ret to `-1`, then have another here to check it.  why not just include this code above in the `if (!addr){` block?